### DISCUSSION TITLE
Serving large files that are generated on-the-fly

### DISCUSSION THREAD

**Discussion by iosofter** [cesanta/mongoose#2504] [Ask questions, get answers] [answered]
I have an application is running on an ESP32 that I'd like to run on an RP2040 with Mongoose. It generates on-the-fly files that are larger than the available RAM size (e.g. a 500K binary or base64 file), and these are displayed using Javascript code running in the browser, and also some standalone applications.

I have looked at your 'huge response' example, and this seems to involve a javascript negotiation between client & server with regard to the transmission of data chunks; unfortunately this is not backwards-compatble with the existing client applications, which effectively do a 'GET data', and expect to receive the whole file.

Can you advise how this might be achieved? In case you wondered about the application, it is [here](https://iosoft.blog/edla)

**Comment by scaprile** [comment, marked as answer]
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation),  [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.

https://mongoose.ws/documentation/tutorials/huge-response/
https://mongoose.ws/documentation/tutorials/http-server/
https://mongoose.ws/documentation/tutorials/http-server/#how-to-iterate-and-send-data <---

Please read https://mongoose.ws/documentation/#best-practices , especially the note
